packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -91,11 +94,12 @@ export type UpdateAndGetHeadOpt =
 export class ForkChoice implements IForkChoice {
   irrecoverableError?: Error;
   /**
-   * Votes currently tracked in the protoArray
-   * Indexed by validator index
-   * Each vote contains the latest message and previous message
+   * Votes currently tracked in the protoArray. Instead of tracking a VoteTracker of currentIndex, nextIndex and epoch,
+   * we decompose the struct and track them in 3 separate arrays for performance reason.
    */
-  private readonly votes: VoteTracker[] = [];
+  private readonly voteCurrentIndices: VoteIndex[] = [];
+  private readonly voteNextIndices: VoteIndex[] = [];
+  private readonly voteNextEpochs: Epoch[] = [];
```

Comment:
```suggestion
  private readonly voteCurrentIndices: VoteIndex[];
  private readonly voteNextIndices: VoteIndex[];
  private readonly voteNextEpochs: Epoch[];
```
Initialized in the constructor